app/helpers/items_helper.rb

```diff
@@ -1,11 +1,8 @@
 module ItemsHelper
   def bought_by item, user
-    orders = user.orders
-    line_items = orders.map { |order| order.line_item }
-    items = line_items.map { |line_item| line_item.item }
-    if items.include? self
-      return true
-    end
-    return false
+    return false unless user
+    items = user.orders.where(status: 'completed').
+      flat_map(&:items)
+    return items.include? item
```

Comment:
flat_map allows us to collect objects from an array into another array, but has the added benefit of flattening the array. Let's say you type this instead:
```ruby
user.orders.where(status: 'completed').map(&:items)
```
The result would be an array of arrays:
```
[[Item1, Item2],[Item5,Item7]]
```
Instead of 
```
[Item1, Item2,Item5,Item7]
```